rust/src/types.rs

```diff
@@ -2851,3 +2892,818 @@ impl ConstantData {
+        unsafe fn from_range<T>(value: BNPossibleValueSet) -> Vec<ValueRange<T>> {
+            core::slice::from_raw_parts(value.ranges, value.count)
+                .iter()
+                .copied()
+                .map(|range| ValueRange::from_raw(range))
+                .collect()
+        }
+        let from_sets = |value: BNPossibleValueSet| {
+            unsafe { core::slice::from_raw_parts(value.valueSet, value.count) }
+                .iter()
+                .copied()
+                .collect()
+        };
+        use BNRegisterValueType::*;
+        match value.state {
+            UndeterminedValue => Self::UndeterminedValue,
+            EntryValue => Self::EntryValue { reg: value.value },
+            ConstantValue => Self::ConstantValue { value: value.value },
+            ConstantPointerValue => Self::ConstantPointerValue { value: value.value },
+            StackFrameOffset => Self::StackFrameOffset {
+                offset: value.value,
+            },
+            ConstantDataValue => Self::ConstantDataValue {
+                value_type: ConstantDataType::Value,
+                value: value.value,
+            },
+            ConstantDataZeroExtendValue => Self::ConstantDataValue {
+                value_type: ConstantDataType::ZeroExtend,
+                value: value.value,
+            },
+            ConstantDataSignExtendValue => Self::ConstantDataValue {
+                value_type: ConstantDataType::SignExtend,
+                value: value.value,
+            },
+            ConstantDataAggregateValue => Self::ConstantDataValue {
+                value_type: ConstantDataType::Aggregate,
+                value: value.value,
+            },
+            SignedRangeValue => Self::SignedRangeValue {
+                offset: value.value,
+                ranges: from_range(value),
+            },
+            UnsignedRangeValue => Self::UnsignedRangeValue {
+                offset: value.value,
+                ranges: from_range(value),
+            },
+            LookupTableValue => {
+                let raw_tables = unsafe { core::slice::from_raw_parts(value.table, value.count) };
+                let raw_from_tables = |i: &BNLookupTableEntry| unsafe {
+                    core::slice::from_raw_parts(i.fromValues, i.fromCount)
+                };
+                let tables = raw_tables
+                    .iter()
+                    .map(|table| LookupTableEntry {
+                        from_values: raw_from_tables(table).to_vec(),
+                        to_value: table.toValue,
+                    })
+                    .collect();
+                Self::LookupTableValue { tables }
+            }
+            ReturnAddressValue => Self::ReturnAddressValue {
+                values: from_sets(value),
+            },
+            InSetOfValues => Self::InSetOfValues {
+                values: from_sets(value),
+            },
+            ImportedAddressValue => Self::ImportedAddressValue,
```

Comment:
Unfinished? These should all have values. `NotInSetOfValues` has the same inner value as `InSetOfValues`